### PR TITLE
COM-3573 B2B P2: Filter B2B account dropdown by Seller

### PR DESCRIPTION
**Pull request by ghost**
Filter the B2B account dropdown based on the Seller. Only B2B accounts a Seller is assigned to should be available. This applies to searching/filtering as well.

- Refactor the mozuPaginatedSearchableDropdown options to accept a single `filter` string with all static filters defined.
- Add the Seller's user ID as a default filter using the `salesrep.userid` filter.
- Add a configuration for the sort direction on the mozuPaginatedSearchableDropdown.

### CODE REVIEW

**Comment by sagarraut0007** [comment]
Changes looks good to me

File: scripts/widgets/mz-search-pagination-dd.js [hunk 1]
```diff
@@ -20,7 +20,11 @@ define(["jquery", "underscore"], function ($, _) {
             "(": "^(",
             ")": "^)"
         };
-        var dataSet = new options.model({ pageSize: options.pageSize, filter: options.filterKey + ' ' + options.filterOption1 + ' ' + options.filterValue, sortBy: options.textField + ' ' + 'asc' });
+        var filterUserIdStr = "";
+        if (options.userId) {
+            filterUserIdStr += "salesrep.userid" + ' ' + options.filterOption1 + ' ' + options.userId + " and ";
+        }
+        var dataSet = new options.model({ pageSize: options.pageSize, filter: filterUserIdStr +  options.filterKey + ' ' + options.filterOption1 + ' ' + options.filterValue, sortBy: options.textField + ' ' + 'asc' });
```
Comment (ghost) [inline scripts/widgets/mz-search-pagination-dd.js]: This is a generic component so we can't hardcode anything inside it. Make your changes using the config options for the component. You will need to revert all your changes in this file.

File: scripts/modules/b2b-account/quotes.js [hunk 2]
```diff
@@ -154,7 +155,8 @@ define([
                         pageSelector: 'selectb2bAccount',
                         optionalFilter: 'and',
                         filterKey: 'isActive',
-                        filterValue: 'true'
+                        filterValue: 'true',
+                        userId: userId
```
Comment (ghost) [inline scripts/modules/b2b-account/quotes.js]: We can't "hardcode" anything like this. This component needs to be generic and work for _any mozu type or any set of filters_. Find another solution that is generic.

Work with Bhushan if you have questions about how this component works. He wrote it.

**Review by ghost — CHANGES_REQUESTED** [review]

File: scripts/widgets/mz-search-pagination-dd.js (hunk 1 above)
Comment (SharmaBhushan) [reply scripts/widgets/mz-search-pagination-dd.js]: Done.

File: scripts/modules/b2b-account/quotes.js (hunk 2 above)
Comment (SharmaBhushan) [reply scripts/modules/b2b-account/quotes.js]: Done

**Review by ghost — CHANGES_REQUESTED** [review]
I still don't think this fix matches my expectations. While the filter for `salesrep.userid` is now configurable this component is not being treated as the generic component it is. There should be no concept of a sales rep and associated filter.

Ideally we would fix this to accept a single generic `filter` parameter that specifies all the filters or a collection of filters and their values.

Let's meet separately and discuss this in more detail.

**Review by ghost — APPROVED** [review]
Thanks for making those changes. I think it's ready to go now.

**Pull request merged**